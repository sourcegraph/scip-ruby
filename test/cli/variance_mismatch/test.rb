# typed: true

module Parent
  extend T::Generic
  Elem = type_member(:out)
end

module Child
  extend T::Generic
  include Parent
  Elem = type_member(:in)
end
