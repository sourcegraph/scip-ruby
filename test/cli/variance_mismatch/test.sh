#!/usr/bin/env bash

main/sorbet --silence-dev-message test/cli/variance_mismatch/test.rb 2>&1
