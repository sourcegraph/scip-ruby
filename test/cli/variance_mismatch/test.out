test/cli/variance_mismatch/test.rb:11: Type variance mismatch for `Elem` with parent `Parent`. Child `Child` should be `:out` or invariant, but it is `:in` https://srb.help/5015
    11 |  Elem = type_member(:in)
          ^^^^^^^^^^^^^^^^^^^^^^^
    test/cli/variance_mismatch/test.rb:5: Parent `Parent` declared here
     5 |  Elem = type_member(:out)
          ^^^^^^^^^^^^^^^^^^^^^^^^
Errors: 1
